libs/plugins/flow-client/src/lib/sub-flow/sub-flow.component.ts

```diff
@@ -1,59 +1,64 @@
-import {
-  AfterViewInit,
-  Component,
-  EventEmitter,
-  Input,
-  OnInit,
-  Output,
-  ViewChild,
-} from '@angular/core';
-import { BsModalComponent } from 'ng2-bs3-modal';
+import { Component, OnInit } from '@angular/core';
 
 import { Resource } from '@flogo-web/core';
-
+import { ModalInstance } from '@flogo-web/lib-client/modal';
+import { LocalSearch, makeLocalSearchFactory } from '@flogo-web/lib-client/search';
+import { Observable } from 'rxjs';
 import { FlogoFlowService as FlowsService } from '../core/flow.service';
+import { FlowResource } from '../core/interfaces';
+
+export interface SubflowSelectionParams {
+  appId: string;
+  currentFlowId: string;
+}
 
 @Component({
   selector: 'flogo-flow-sub-flow',
   templateUrl: 'sub-flow.component.html',
   styleUrls: ['sub-flow.component.less'],
+  providers: [
+    {
+      provide: LocalSearch,
+      useFactory: makeLocalSearchFactory({
+        matchFields: ['name', 'description'],
+      }),
+    },
+  ],
 })
-export class SubFlowComponent implements AfterViewInit, OnInit {
-  @Input()
-  appId: string;
-  @Input()
-  currentFlow: string;
-  @Output()
-  flowSelected: EventEmitter<Resource | string> = new EventEmitter<Resource | string>();
-
-  @ViewChild('listModal', { static: true }) modal: BsModalComponent;
-  flowsList: Resource[];
+export class SubFlowComponent implements OnInit {
+  private appId: string;
+  private readonly currentFlowId: string;
+  flows$: Observable<FlowResource[]>;
 
-  constructor(private flowService: FlowsService) {}
+  constructor(
+    private flowService: FlowsService,
+    private search: LocalSearch<FlowResource>,
+    private modal: ModalInstance<SubflowSelectionParams>
+  ) {
+    const { appId, currentFlowId } = this.modal.data;
+    this.appId = appId;
+    this.currentFlowId = currentFlowId;
+  }
 
   ngOnInit() {
-    this.flowService.listFlowsForApp(this.appId).subscribe(flows => {
-      this.flowsList = flows.filter(flow => flow.id !== this.currentFlow);
+    this.flows$ = this.search.list$;
```

Comment:
Do you mean in the template? I guess but I decided against it because I didn't want to do that because then I would need to make the `search` service public and it didn't make sense to me to expose it like that.